src/rulesets/Base/sort.jl

```diff
@@ -33,3 +37,54 @@ function rrule(::typeof(sort), xs::AbstractVector; kwargs...)
     end
     return ys, sort_pullback
 end
+
+#####
+##### `sortslices`
+#####
+
+function rrule(::typeof(sortslices), x::AbstractArray{<:Number}; dims::Integer, kw...)
+    p = sortperm(collect(eachslice(x; dims=dims)); kw...)
+    inds = ntuple(d -> d == dims ? p : (:), ndims(x))
+    function sortslices_pullback(dy)
+        # No actual need to zero this, and if you didn't, then you could widen eltype
+        # Also, you could use similar(dy) here not x, same size?
+        dx = _zerolike_writeat(x, unthunk(dy), (), inds...)
+        return (NoTangent(), ProjectTo(x)(dx))
+    end
+    return x[inds...], sortslices_pullback
+end
+
+#####
+##### `unique`
+#####
+
+function rrule(::typeof(unique), x::AbstractArray{<:Number}; dims=:)
+    axes_x = axes(x)
+    y = unique(x; dims=dims)  # accepts only dims=: or dims::Integer
+    function unique_pullback(dy_raw)
+        dy = unthunk(dy_raw)
+        if length(x) == length(y)
+            # Short-circuit for the case of all unique, since `mask` is fairly expensive:
+            dx = reshape(dy, axes_x)
+            return (NoTangent(), ProjectTo(x)(dx))
+        end
+        if dims isa Colon
+            xs, ys = vec(x), y
+        else
+            xs, ys = collect(eachslice(x; dims=dims)), collect(eachslice(y; dims=dims))
```

Comment:
There is an issue open on BlueStyle to remcomment against this
https://github.com/invenia/BlueStyle/issues/80

If we are going to do this then how do you feel about:
```suggestion
            xs, ys = collect.(eachslice((x, y); dims=dims))
```

Reply:
We could avoid this for style. But I think the broadcast is confusing, and perhaps you do too, because I also think it's missing an easy-to-miss dot:
```
julia> collect.(eachslice((x, y); dims=1))
ERROR: MethodError: no method matching eachslice(::Tuple{Matrix{Float64}, Matrix{Float64}}; dims=1)

julia> collect.(eachslice.((x, y); dims=1))
(SubArray{Float64, 1, Matrix{Float64}, Tuple{Int64, Base.Slice{Base.OneTo{Int64}}}, true}[[0.5119304534786525, 0.6182654562598278, 0.16701230957752622], [0.7959010118362386, 0.9477852004109513, 0.3864
```